### PR TITLE
Post/site/template editors: clear selection on iframe html element click, fix bottom click redirect

### LINKED ISSUE #31369: Writing flow/FSE: Clicking in the canvas padding doesn't always deselect
## Description

When you select a block, it's important to be able to click outside to deselect it. That generally works, but if you click right on the edge, the cursor turns to the text caret, and you actually set focus in the bottom appender:

click

This is very jarring when it happens in the post editor, as you lose your place. But more importantly in the site editor which uses all the space available, it can make it very hard to deselect any blocks at all.

The clickable space that focuses the last appender is actually the padding on `editor-styles-wrapper` — click that, and you scroll to the end of the page:

Screenshot 2021-04-30 at 09 52 14

How much padding is there will vary from theme to theme, of course, but simply clicking there shouldn't make you lose your place.

## Step-by-step reproduction instructions

Go to the post or site editor, insert a bunch of content.

- Select a block and click outside, left or right to deselect it.
- Try again, but carefully move the cursor to the edge of the screen and see if it changes to the text editing caret. Click there, and observe how you're now focused in the bottom appender.

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Fixes #31369. Also fixes the selection clearer below the fold for FSE. It need to be active on the `html` element as well, so I built it in the Iframe component.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by jasmussen** [comment]
Thank you for working on this. The effect on the post editor is subtle but amazing:

here

Clicking left or right of a block deselects, even if you click in the padding of the body element.

It's not working quite as intended in the site editor, alas. I think it's the code that looks for the bottom block, which comes very early on archive pages in the site editor, meaning if you scroll just a little bit, you basically can't deselect:

site editor

I understand the problem is related to the "click redirect" — the feature where if you click below the last block it sets focus on the last block. If we can keep that in the post editor, I still think that would be ideal. But if we have to remove that, it _might_ still be worth it. Ultimately I thin it's more important to be able to deselect blocks than it is to easily select the last one.

**Comment by ellatrix** [comment]
> I understand the problem is related to the "click redirect"

@jasmussen No, this particular problem in the site editor is not related to the click redirect, it is not added there at all. The problem is that the selection clearer is active on the body element, but not the html element, and the body doesn't extend all the way to the bottom. I've fixed the problem now by also adding it to the html element.

File: packages/edit-post/src/components/visual-editor/style.scss [hunk 1]
```diff
@@ -33,12 +33,6 @@
 	// color doesn't show through.
 	background-color: $white;
 	flex: 1;
-
-	cursor: text;
-
-	> * {
-		cursor: auto;
-	}
```
Comment (ellatrix) [inline packages/edit-post/src/components/visual-editor/style.scss:41]: We need to remove the cursor style as we can't apply it to only the bottom padding of an element.

**Comment by jasmussen** [comment]
Currently seeing these two:

post editor

site editor

Both of these work really well for me. The site editor has a few small issues but those are unrelated to this PR. I'm a fan, would love to see this land. What's needed to land it?

**Comment by ellatrix** [comment]
It works great, so let's merge. There's a small issue with FSE that exists in trunk too where you have to click twice above the first block to clear selection, but I'll have to look into that separately.

**Comment by Mamaduka** [comment]
@ellatrix, @jasmussen, I would appreciate your feedback on this revert PR - #73753.

**Pull request merged**